src/Swipeable.js

```diff
@@ -1,106 +1,123 @@
-// Based on https://github.com/john-doherty/swiped-events
-import { PureComponent } from "react";
+import { useEffect, useRef, forwardRef, useCallback } from "react";
 import PropTypes from "prop-types";
 
-export default class Swipeable extends PureComponent {
-  constructor(props) {
-    super(props);
+const Swipeable = forwardRef(function Swipeable(props, ref) {
```

Comment:
@Yaxian looks better now.

What would happen if we completely remove the `forwardRef` and `ref` usage? 

So exporting something like:
```javascript
function Swipeable(props) {
  ...
}
```

I have update this [codesandbox](https://codesandbox.io/s/testing-version-with-hooks-ipt2k?file=/src/Swipeable.js) with your latest code.

Reply:
Tests will fail and `function component` can not pass `ref` props.

Reply:
Ok, does that mean any consumer component will break? 
Using it like the way it is now
```javascript
<Swipeable
        onSwipeLeft={this.handleSwipeLeft}
        onSwipeRight={this.handleSwipeRight}
      >
        {innerRef => (
          <div ref={innerRef}>
            ...
          </div>
        )}
      </Swipeable>
```
isn't `innerRef` going to be passed correctly without `forwardRef ` usage?

I briefly tested in the above codesandbox without `ref` and it seems to work fine. Can you confirm that?

Reply:
We can not pass `ref` directly to `Swipeable`:
```
<Swipeable ref={ref} onSwipeLeft={() => {}}>
   {innerRef => (
          <div ref={innerRef}>
            ...
          </div>
        )}
</Swipeable>
```
if we do not use `forwardRef` to wrap a `function component`.

`innerRef ` still works fine.

Reply:
Great, I think it's not necessary to change the user facing API of Swipeable for this rewrite, so let's keep the usage of `innerRef` and skip `forwardRef`.

Reply:
the tests will fail  if we do not use forwardRef, cause we can not get the instance and its methods of `function component` like `class component`

Reply:
That's true. Let's keep the `forwardRef ` then. 

I'll merge the MR soon.